### ISSUE TITLE
[HOLD for payment 2023-03-01] [$4000] Few currency symbols (specially symbols after WST) are not correctly displayed on mWeb chrome on android

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14307]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Open any report
3. Click on plus and open Request Money/Send Money
4. Scroll to bottom and see the currencies below WST on mWeb chrome

## Expected Result:
Symbol for XAF should be FCFA, XCD symbol should be EC$, XOF symbol should be CFA and XPF symbol should be CFPF

## Actual Result:
Symbols are not correctly displayed, they are same as name of currency for XAF, XCD, XOF, XPF on mWeb chrome android

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] Android / Chrome

**Version Number:** 1.2.54-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/212432110-97a16406-7d0e-4b89-8a6b-4a013ba7babc.mp4

https://user-images.githubusercontent.com/43996225/212432163-88f6a260-5fb3-4eb1-9e43-f062e6d8cd81.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673630821293879

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c6b2d46695b607ed</li>
        <li>Upwork Job ID: 1617777751000506368</li>
        <li>Last Price Increase: 2023-02-08</li>
    </ul>
</details>

**Comment by slafortune** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [insert C/C+ member] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:
- [ ] [insert internal engineer and C+] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/12301
- [ ] [insert internal engineer and C+] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A - @Beamanator the author is aware and part of the discussion when this regression was caught. 
- [ ] [insert internal engineer and C+] A discussion in #contributor-plus has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [insert BZ member] Payment has been made to the issue reporter (if applicable)
- [ ] [insert BZ member] Payment has been made to the contributor that fixed the issue (if applicable)
- [ ] [insert BZ member] Payment has been made to the contributor+ that helped on the issue (if applicable) - N/A we merged this one immediately as a breaking change before becoming a deploy blocker.

**Comment by slafortune** [comment]
Is this something that could be fixed by an external contributor @techievivek ?

**Comment by techievivek** [comment]
Looking into now.

**Comment by techievivek** [comment]
Weird. It just seems to be happening on Chrome(Android). This can be worked externally.

**Comment by techievivek** [comment]
Great work @Prince-Mendiratta, loved the breakdown that you posted above.

CC @mananjadhav do you mind reviewing the above proposal? IMO I would say solution 1 would be the best here, but curious to hear your thoughts and understand why we never used polyfill for all the platforms. Is it because we only see a need for it now?

**Comment by mananjadhav** [comment]
@techievivek I think we should post this on the slack?  Quick read I agree with Solution 1, but I am not sure why we didn't use polyfills. Also IIRC (I could be wrong), we had polyfills earlier? and we removed them later? Hence better we check on Slack?

**Comment by techievivek** [comment]
Discussing here: https://expensify.slack.com/archives/C01GTK53T8Q/p1675152607405109

**Comment by techievivek** [comment]
@mananjadhav Could you kindly review the updated proposal?

**Comment by mananjadhav** [comment]
Okay I went through the discussion and proposal. @Prince-Mendiratta I checked but your proposal didn't work for me. I am checking this on Pixel 5 Android emulator chrome.

<img width="365" alt="Screenshot 2023-02-02 at 11 21 22 AM" src="https://user-images.githubusercontent.com/3069065/216242887-d905b0b2-8f6f-4eee-af84-0ccf74298426.png">

Also I think we should just import the `locale-data`? and not the polyfill?

**Comment by techievivek** [comment]
Not overdue, I will have a look at this by the EOD.

**Comment by mananjadhav** [comment]
I don't think replacing `format-js` with `intl` is right solution here. Instead if there's any fix required on `format-js` or it's locale-data, let's do that so that the community can also benefit from it.

**Comment by techievivek** [comment]
Catching up here as I was ooo, @Prince-Mendiratta, Where have you asked for the help? Can you share the link? I have also put the issue on HOLD until we have some way forward.

**Comment by techievivek** [comment]
Thanks for the link @Prince-Mendiratta. We have put the issue on the hold until we have some response from the format js team.

**Comment by techievivek** [comment]
@Prince-Mendiratta Thanks for taking the lead.

One quick question that I want to clarify: The reason mWeb(Chrome) doesn't return the correct symbol for `XAF` is because API implementation is different for mWeb(Chrome) or is it because it's missing the locale data?

**Comment by techievivek** [comment]
The solution forward is 
1. We want to use `polyfill-force`(so it polyfills the Intl API, right?).
2. Load the locale data.

We already seem to be doing that for `platformSetup/index.native.js`, so all we are left to do is to use `polyfill-force`?

**Comment by techievivek** [comment]
@Prince-Mendiratta 
>  might have older versions of ICU (International Components for Unicode, responsible for providing the locale data) lib.

I tried reading about it but wasn't sure how this would interfere with the locale data, thanks for the explanation here.

I like the [updated proposal](https://github.com/Expensify/App/issues/14307#issuecomment-1431921542) to conditionally polyfill. Also, looking into the docs they seem to have this [`shouldPolyfill`])(https://formatjs.io/docs/polyfills/intl-numberformat#dynamic-import--capability-detection) method can we not use that?


Ccing @mananjadhav to weigh in his thoughts.

**Comment by techievivek** [comment]
@Prince-Mendiratta, I see. I completely missed the underlying logic. Looks good to me, I will wait for @mananjadhav to give it a look. Great work 🙌

**Comment by mananjadhav** [comment]
Okay I went through the logic, and I think that would work. But I think we should move this to a separate file and not clutter the `setup/index.js`.

**Comment by techievivek** [comment]
@mananjadhav, Can you please officially review the proposal so I can assign @Prince-Mendiratta for the job? Thanks

**Comment by mananjadhav** [comment]
@techievivek  I reviewed the proposal [here](https://github.com/Expensify/App/issues/14307#issuecomment-1431921542) and I think we're good with @Prince-Mendiratta's proposal. All the kinks related to the code styling can be handled in the PR stage.

**Comment by techievivek** [comment]
Cool, cool.

**Comment by MelvinBot** [comment]
📣 @Prince-Mendiratta You have been assigned to this job by @techievivek!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by techievivek** [comment]
Alright, I have assigned the job to @Prince-Mendiratta.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.75-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15245


If no regressions arise, payment will be issued on 2023-03-01. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
📣 @MelvinBot! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mananjadhav / @techievivek] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mananjadhav / @techievivek] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mananjadhav / @techievivek] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@Prince-Mendiratta] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@slafortune] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by mananjadhav** [comment]
>  The currency symbol displayed in the IOU transaction screen is just the string passed around.

Agreed. Just for the sake of completing the whole workflow.

**Comment by mananjadhav** [comment]
@techievivek Quick bump on this one.

**Comment by mananjadhav** [comment]
@techievivek did you get a chance to look at the test steps?

@slafortune this is ready for payout today and wanted to confirm that the PR was merged within 3 business days (excluding weekends), so hoping it is eligible for the timeline bonus.

**Comment by techievivek** [comment]
@mananjadhav Yeah, it looks good to me. How do we go about adding it? Any idea?

**Comment by mananjadhav** [comment]
I am not sure. I did see a post on C+ channel about Testrail, but I have never done it here. 

> Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

Based on the checklist, I think @slafortune can help here.

**Comment by techievivek** [comment]
Not overdue, @slafortune will add the regression steps today.

**Comment by techievivek** [comment]
Gentle bump @slafortune to get the [regression test](https://github.com/Expensify/App/issues/14307#issuecomment-1444275610) added to the testRail.

**Comment by slafortune** [comment]
Sorry for my delay everyone - done!

**Comment by slafortune** [comment]
Hmm- I did receive an error but it appeared to go through - let me check!

**Comment by dhanashree-sawant** [comment]
Hi @slafortune, can you check once you are back to work?

**Comment by slafortune** [comment]
Alright -  not sure what I messed up to start with, but looks like we should be good now!  Shout if that isn't the case @dhanashree-sawant @Prince-Mendiratta

**Comment by dhanashree-sawant** [comment]
Hi @slafortune , all good, thanks for payment.
